src/style.rs

```diff
@@ -0,0 +1,162 @@
+use std::borrow::Cow;
+
+/// Rich (formatted) text cell.
+#[derive(Debug, PartialEq, Clone, Default)]
+pub struct RichText {
+    /// Total text, where the different formats are saved in the `formatted` field.
+    text: String,
+    /// List of format annotations, with each having:
+    /// - Length of string section
+    /// - Format of the string section
+    formatted: Vec<(usize, FontFormat)>,
+}
+
+impl RichText {
+    /// Create new empty rich text.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Create an instance with only default plain text.
+    pub fn plain(s: String) -> Self {
+        let len = s.len();
+        Self {
+            text: s,
+            formatted: vec![(len, FontFormat::default())],
+        }
+    }
+
+    /// Is this empty?
+    #[inline]
+    pub fn is_empty(&self) -> bool {
+        self.text.is_empty()
+    }
+
+    /// Is this just plain text?
+    #[inline]
+    pub fn is_plain(&self) -> bool {
+        self.formatted.iter().all(|(_len, f)| f.is_default())
+    }
+
+    /// Get the full text value without formatting.
+    #[inline]
+    pub fn text(&self) -> &String {
```

Comment:
`&str` ?

Reply:
In general, it is better to return the type it is, but accept a str or slice in parameters, as that allows the greatest freedom for the user.
Returning a &str restricts the user and does not allow to call methods from String (not sure if there are any interesting), but the user can easily get a &str. The other way around, it is not possible to get a &String from &str without an additional allocation.

But of course it might be annoying to have to convert to &str, so I have no hard feelings, which one should I pick?

Reply:
I'd naturally pick `&str` for public api to forward proof it (avoid leaking technical details and allows an eventual other type to be used wo breaking change). I feel like most people won't need the `String` (there is already an `into_text` if necessary)